public/app/percona/backup/components/BackupInventory/RestoreBackupModal/RestoreBackupModal.tsx

```diff
@@ -38,12 +42,111 @@ export const RestoreBackupModal: FC<RestoreBackupModalProps> = ({
 }) => {
   const styles = useStyles(getStyles);
   const initialValues = useMemo(() => (backup ? toFormProps(backup) : undefined), [backup]);
+
+  const [selectedTimerange, setSelectedTimerange] = useState<Timeranges>();
+  const [selectedTimerangeFromDatepicker, setSelectedTimerangeFromDatepicker] = useState<DateTime>();
+  const [selectedDay, setSelectedDay] = useState<Date | undefined>();
   const handleSubmit = ({ serviceType, service }: RestoreBackupFormProps) => {
-    if (backup) {
+    if (backup && selectedTimerangeFromDatepicker) {
       const serviceId = serviceType === ServiceTypeSelect.SAME ? backup.serviceId : service.value;
-      onRestore(serviceId || '', backup.id);
+      onRestore(serviceId || '', backup.id, selectedTimerangeFromDatepicker.toISOString());
     }
   };
+  const calculateDisableHours = useCallback(() => {
+    const disabledHours = [];
+
+    if (selectedTimerange) {
+      const { startTimestamp, endTimestamp } = selectedTimerange;
+      for (let i = 0; i < 24; i++) {
+        if (moment(selectedDay).isSame(startTimestamp, 'date')) {
+          if (i < moment(startTimestamp).hours()) {
```

Comment:
Question: our loops goes from 0 to 23. Does `.hours()` return from 0 to 23 as well?